fame/core/module_dispatcher.py

```diff
@@ -172,6 +214,23 @@ def add_processing_module(self, module):
                     for source_type in iterify(module['acts_on']):
                         self._add_trigger(self._triggers, "_generated_file(%s)" % source_type, module)
 
+    def add_preloading_module(self, module):
+        def add_one(acts_on, module):
+            if acts_on not in self._modules['Preloading']:
+                self._modules['Preloading'][acts_on] = []
+
+            self._modules['Preloading'][acts_on].append(module)
+
+        m = get_class(module['path'], module['class'])
+        if m:
+            m.info = module
+
+            self._add_module_options(module)
+
+            if module['acts_on']:
+                for source_type in iterify(module['acts_on']):
+                    add_one(source_type, m)
```

Comment:
Why so complicated when the only type is `hash` ?

Reply:
You are right. See my other comment on simplifiying PreloadingModules.